crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +137,207 @@ impl Persistence {
+            .await
+            .context("load solver competition")?
+            .ok_or(error::Solution::SolverCompetition(anyhow::anyhow!(
+                "missing solver competition"
+            )))?;
+        let competition: model::solver_competition::SolverCompetitionDB =
+            serde_json::from_value(solver_competition.json).map_err(|_| {
+                error::Solution::SolverCompetition(anyhow::anyhow!("json conversion"))
+            })?;
+        let winning_solution =
+            competition
+                .solutions
+                .last()
+                .ok_or(error::Solution::SolverCompetition(anyhow::anyhow!(
+                    "empty solutions"
+                )))?;
+        let mut orders = HashMap::new();
+        for order in winning_solution.orders.iter() {
+            match order {
+                model::solver_competition::Order::Colocated {
+                    id,
+                    sell_amount,
+                    buy_amount,
+                } => {
+                    orders.insert(
+                        domain::OrderUid(id.0),
+                        competition::TradedAmounts {
+                            sell: (*sell_amount).into(),
+                            buy: (*buy_amount).into(),
+                        },
+                    );
+                }
+                model::solver_competition::Order::Legacy {
+                    id: _,
+                    executed_amount: _,
+                } => {
+                    return Err(error::Solution::SolverCompetition(anyhow::anyhow!(
+                        "Legacy order"
+                    )))
+                }
+            }
+        }
+        let mut prices = HashMap::new();
+        for (token, price) in winning_solution.clearing_prices.clone().into_iter() {
+            prices.insert(
+                token.into(),
+                domain::auction::Price::new(price.into())
+                    .map_err(|_| error::Solution::Price(domain::auction::InvalidPrice))?,
+            );
+        }
+        Ok(competition::Solution::new(0, winner, score, orders, prices))
+    }
 }
 
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
     #[error("failed to read data from database")]
     DbError(#[from] anyhow::Error),
-    #[error(transparent)]
-    Price(#[from] domain::auction::InvalidPrice),
+}
+
+pub mod error {
+    use super::*;
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum Auction {
+        #[error("failed to read data from database")]
+        DbError(#[from] anyhow::Error),
+        #[error("failed dto conversion from database")]
+        DbConversion(&'static str),
+        #[error(transparent)]
+        Price(#[from] domain::auction::InvalidPrice),
+        #[error("deadline not found in the database")]
+        MissingDeadline,
+        #[error("quote not found in the database for an existing order")]
+        MissingQuote,
```

Comment:
Should this contain the order id for which the quote is missing to make it easier to debug this issue when oncall sees this error?